test-basic/documents-data-movement-readAll.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const should = require('should');
+const testconfig = require('../etc/test-config.js');
+const marklogic = require('../');
+const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+const Stream = require('stream');
+const streamToArray = require('stream-to-array');
+const fs     = require('fs');
+
+let readable = new Stream.Readable({objectMode: true});
+let uris = new Stream.Readable();
+let urisList = [];
+
+describe('data movement readAll', function() {
+    before(function (done) {
+        for(let i=0; i<10000; i++) {
+            const temp = {
+                uri: '/test/dataMovement/requests/readAll/'+i+'.json',
+                contentType: 'application/json',
+                content: {['key '+i]:'value '+i}
+            };
+            readable.push(temp);
+            uris.push(temp.uri);
+            urisList.push(temp.uri);
+
+        }
+        readable.push(null);
+        uris.push(null);
+        readable.pipe(dbWriter.documents.writeAll());
+        setTimeout(()=>{done();}, 3000);
+    });
+
+    after((function(done){
+        dbWriter.documents.remove(urisList)
+            .result(function(response){
+                done();
+            })
+            .catch(err=> done(err))
+            .catch(done);
+    }));
+
+    it('should readAll documents with empty options', function(done){
+
+        uris.pipe(dbWriter.documents.readAll()).pipe(fs.createWriteStream('file.txt'));
```

Comment:
I suspect that a readable stream can only be read once.

If so, it may be necessary to construct the urisList from the Stream.Readable separately for the initialization and the test.